Maestro/src/Microsoft.DotNet.Maestro/Handlers/VsoBuildHandler.cs

```diff
@@ -38,7 +38,8 @@ public static ISubscriptionHandler TryCreate(JObject action, Subscription subscr
                     VsoProject = action["vsoProject"].ToString(),
                     BuildDefinitionId = action["buildDefinitionId"].Value<int>(),
                     SourceBranch = GetSourceBranch(subscription),
-                    VsoParameters = VsoParameterGenerator.GetParameters(subscription.ActionArguments, parameterSubstitutions)
+                    VsoParameters = VsoParameterGenerator.GetParameters(subscription.ActionArguments, parameterSubstitutions),
+                    VsoService = new VsoService(action["vsoInstance"].ToString())
```

Comment:
This should be in `Execute`. Here, it'll work if there's no delay set, but if there's a delay, the class is JSON serialized and sent off to be executed later by the WebJob, where it's deserialized from JSON and this property is lost:

https://github.com/dotnet/versions/blob/95511b94d68bdd058b0c0574e7f2e4d8f39d1068/Maestro/src/Microsoft.DotNet.Maestro.WebJob/Functions.cs#L33-L34

(This is also causing the CI issue.)

Reply:
@dagood I think the CI issue is actually caused by the fact that it returned an empty string for credentials.

Reply:
I mean that at a higher level, CI shouldn't be trying to get credentials and moving it to `Execute` would fix that.